script/src/validate-monorepo/validation/index.ts

```diff
@@ -18,7 +18,27 @@ export async function* validateMonorepo(): AsyncGenerator<ValidationIssue> {
   const libs = await readdir(join(root, 'libs'));
   const packages = [...services, ...libs, ...appPackages];
 
-  yield* pkgs.checkConfig({ packages: [root, ...packages] });
+  yield* pkgs.checkConfig({
+    packages: [root, ...packages],
+    // It's important that these packages are pinned to a specific version.
```

Comment:
Provocation: Is there any package that this invariant _shouldn't_ apply to?

Reply:
It's a good question. One example is Jest. We don't have much in the way of shared code that uses the Jest API, and it's likely we will want to upgrade Jest one package at a time to make it tractable. I'd be open to making this opt out instead of opt in in the future, but for now I think we should go with this.

Reply:
Sounds good!